### PR TITLE
Adapt Selenium tests for SPI branch

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### What does this PR do?
Add changes to Selenium tests, to make them work with Che SPI

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/5996

### Changelog
Adapt selenium tests for SPI branch

### Release Notes
Selenium tests can be executed for Che SPI branch

### Docs PR
none

### CODE REVIEW

File: selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml
Comment (garagatyi) [inline selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml:579]: Can you elaborate on why some tests are removed from an xml configs and from java classes?

File: selenium/che-selenium-test/pom.xml
Comment (garagatyi) [inline selenium/che-selenium-test/pom.xml]: Can you elaborate on how this dependency appears in runtime?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: It looks like a syntax error. Can you check it?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: This statement is unreachable

**Comment by garagatyi** [comment]
I removed several reviewers because they appear as code owners by a mistake

File: selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml:579]: Those 2 tests are for functionality that is no longer present in SPI branch

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: fixed

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: fixed

File: selenium/che-selenium-test/pom.xml
Comment (mkuznyetsov) [reply selenium/che-selenium-test/pom.xml]: I added it once to prevent NoClassDefFound exception at test executions. But I can't reproduce it anymore, so I'll remove it

**Review by vparfonov — APPROVED** [review]

**Review by skabashnyuk — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java]: Reuse constant from workspace module.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: dev machine should be found using installers OR servers. E.g.
```
.getMachine()
.values()
.stream()
.filter(m -> {
    return InstallerFqn.idInKeyList("org.eclipse.che.ws-agent", m.getInstallers()) || m.getServers().keySet().contains(Constants.SERVER_WS_AGENT_HTTP_REFERENCE);
})
.forEach(m -> m.getAttributes().put("memoryLimitBytes", Long.toString(convertToByte(memory, memoryUnit))));
```
BTW we have pretty the same logic in workspace module. It is in class WsAgentMachineFinderUtil.
So I would suggest to add new method to this class and use it here.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: Add nullable annotation, please

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: Please, use WsAgentMachineFinderUtil and get by port inside server object as now key in servers map is not port but reference. And ServerConfig should be used instead of Server

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java
Comment (musienko-maxim) [inline selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java:116]: Can we replace pubic interface on public static constans in the place-? And syntax err.: MAXIMAZE-> MAXIMIZE, MINIMAZE -> MINIMIZE

File: selenium/che-selenium-test/pom.xml
Comment (dmytro-ndp) [inline selenium/che-selenium-test/pom.xml:109]: We need it at runtime to have GitHubKey DTO implementation, to avoid an error:
```
prepare(org.eclipse.che.selenium.git.WorkingWithPullConflictsTest) Time elapsed: 32.284 sec <<< FAILURE!
java.lang.IllegalArgumentException: Provider of implementation for DTO type interface org.eclipse.che.plugin.github.shared.GitHubKey isn't found
at org.eclipse.che.dto.server.DtoFactory.getDtoProvider(DtoFactory.java:396)
at org.eclipse.che.dto.server.DtoFactory.createDto(DtoFactory.java:181)
at org.eclipse.che.dto.server.DtoFactory.newDto(DtoFactory.java:94)
at org.eclipse.che.selenium.core.client.TestGitHubServiceClient.uploadPublicKey(TestGitHubServiceClient.java:71)
at org.eclipse.che.selenium.git.WorkingWithPullConflictsTest.prepare(WorkingWithPullConflictsTest.java:116)
```

File: selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml
Comment (dmytro-ndp) [reply selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml:579]: Please, don't forget to delete those tests CreatePackagesAndFilesWithHelpIconCreateTest, DeleteFilesPackagesAndProjectWithHelpDeleteIconTest  from test suites https://github.com/codenvy/codenvy/blob/master/selenium/codenvy-selenium-test/src/test/resources/suites/CodenvyOnpremSuite.xml#L522l and https://github.com/codenvy/saas/blob/master/selenium/saas-selenium-test/src/test/resources/suites/CodenvySaasSuite.xml#L531 after the merge changes into the master.

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java]: It is not likely, but still possible to get NPE here in case machine doesn't have `SERVER_WS_AGENT_HTTP_REFERENCE`.
Can you check for nullness for this case before retrieving the URL?

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (garagatyi) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: This method looks suspicious in SPI branch. It is probable that we should remove usage of it. Can you deprecate it to show that it should not be used in future?

File: selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/test/resources/suites/CheSuite.xml:579]: we can do it if and when we will be testing Codenvy on SPI

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: Added

File: selenium/che-selenium-test/pom.xml
Comment (garagatyi) [reply selenium/che-selenium-test/pom.xml:109]: Can you guys decide on which way is correct and make a change according to that?

File: selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java
Comment (garagatyi) [reply selenium/che-selenium-test/src/main/java/org/eclipse/che/selenium/pageobject/ProjectExplorer.java:116]: +1

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: This method is renamed to 
```public Server getServerFromDevMachineBySymbolicName(String workspaceId, String serverName)```
 because it is used in that way (getting server by it's name in the map)

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: Ok

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java]: Added and using method from WsAgentMachineFinderUtil now

File: selenium/che-selenium-test/pom.xml
Comment (mkuznyetsov) [reply selenium/che-selenium-test/pom.xml:109]: I didn't manage to reproduce the error regarding to these runtime dependencies. But if we get I'd remove these for now, and return if the problem occurs again

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (garagatyi) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java]: Are you sure that this symbolic name won't change? AFAIK we don't have a policy on how we create servers for exposed ports if the user has not provided his own server.
So, I suggest adding the server into workspace config and use it instead.

File: selenium/che-selenium-test/pom.xml
Comment (garagatyi) [reply selenium/che-selenium-test/pom.xml:109]: Sounds good!

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java]: ok

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestProjectServiceClient.java]: ok

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java
Comment (mkuznyetsov) [reply selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestWorkspaceServiceClient.java]: ok

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (garagatyi) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java:70]: Please, reuse utility for deserialization

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (garagatyi) [inline selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java]: Please, add teardown method which stops and deletes the WS

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java:70]: ok

File: selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java
Comment (mkuznyetsov) [reply selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/gwt/CheckSimpleGwtAppTest.java]: ok

**Review by garagatyi — APPROVED** [review]

**Review by musienko-maxim — APPROVED** [review]

File: selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/WorkspaceDtoDeserializer.java
Comment (tolusha) [inline selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/utils/WorkspaceDtoDeserializer.java:24]: Singleton?

**Review by tolusha — APPROVED** [review]

**Pull request merged**
